File: src/main/java/com/york/asciiartstudio/model/adapters/LiveStreamSource.java
package com.york.asciiartstudio.model.adapters;

public interface LiveStreamSource {
    /**
     *
     * @return the width in pixels of the original video.
     */
    int getWidth();

    /**
     *
     * @return the height in pixels of the original video.
     */
    int getHeight();

    /**
     *
     * @return the frames per second of the original video.
     */
    int getFPS();

    /**
     *
     * @return the current frame (at call time) of the live steam source as an ImageSource.
     */
    ImageSource getCurrentFrame();
}

File: src/main/java/com/york/asciiartstudio/model/adapters/LiveStreamSource.kt
package com.york.asciiartstudio.model.adapters

interface LiveStreamSource {

    /**
     * @return the width in pixels of the original video.
     */
    val width: Int

    /**
     * @return the height in pixels of the original video.
     */
    val height: Int

    /**
     * @return the frames per second of the original video.
     */
    val fps: Int

    /**
     * @return the current frame (at call time) of the live steam source as an ImageSource.
     */
    val currentFrame: ImageSource
}